classes/order/Order.php

```diff
@@ -1607,6 +1607,21 @@ public static function getByReferenceAndEmail($reference, $email)
 
         return new Order($id);
     }
+    
+    public static function getByReferenceEmailAndNumber($reference, $email, $ordernumber)
+    {
+        $sql = '
```

Comment:
Perhaps the same SQL of the previous method can be reused somehow avoiding code duplicate.